src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -5,11 +5,13 @@
 // Changes to this file must follow the http://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+using System.Net.Security;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace System.Net.Quic
 {
-    public sealed partial class QuicConnection : System.IDisposable
```

Comment:
In general we should be deliberate and careful with API changes. We are currently not going through the API approval process for System.Net.Quic, but that means we need to have our own internal process for discussing and agreeing to API changes.

I'd prefer to handle any API changes separately from this PR, if possible. That will avoid delaying this PR while we discuss API changes. Is that possible?

Reply:
> We are currently not going through the API approval process for System.Net.Quic, but that means we need to have our own internal process for discussing and agreeing to API changes.

What do we need to do to make this process more formal then? Throughout the PR, I stated many of the reasons I was going to make an API change. For example, the StartAsync change, I stated the design reasons and decisions for making the change in the PR: https://github.com/dotnet/runtime/pull/427#discussion_r354554666. If this isn't enough process, then let's make sure we agree the next time we make changes to the API.

> I'd prefer to handle any API changes separately from this PR, if possible. That will avoid delaying this PR while we discuss API changes. Is that possible?

It would be difficult to change this PR to not change the API at this point. For example, I added options types to many thing; removing all the pluming there would be counterproductive. 

Can you clarify why it would be better not to change the API for now? I think iterating on the API is fine within these prototype like PRs.

Reply:
> What do we need to do to make this process more formal then?

Good question. For regular API reviews, the process is like this:
(1) File an issue with the proposed API change/addition
(2) Discuss on the issue and drive to consensus
(3) Take it to the weekly API approval board meeting for further discussion and approval or rejection
(4) If approved, owner goes and implements the change

I don't think we need to be nearly that formal, but I do think we should have some process here. I'd suggest something like this:
(1) Post suggested API change to the teams chat
(2) Discuss and drive to consensus
(3) Implement

In other word, basically we should just get an ack from all interested parties before making API changes here.

Reply:
> It would be difficult to change this PR to not change the API at this point. For example, I added options types to many thing; removing all the pluming there would be counterproductive.

Okay, then let's try to review these quickly and get them approved. Sound reasonable?

Reply:
Sure, I'll post them in the teams channel shortly for review.

Reply:
> Can you clarify why it would be better not to change the API for now? 

I want to make sure we're deliberate about API changes. I want to avoid API churn. I want to ensure that any changes we make are intended to be permanent and not just temporary placeholders until we figure it out. I want to ensure we're making API changes that make sense regardless of implementation backend (e.g. msquic) and avoid churn here.